ruma-api-macros/src/api/metadata.rs

```diff
@@ -91,8 +104,16 @@ impl Parse for Metadata {
             method: method.ok_or_else(|| missing_field("method"))?,
             name: name.ok_or_else(|| missing_field("name"))?,
             path: path.ok_or_else(|| missing_field("path"))?,
-            rate_limited: rate_limited.ok_or_else(|| missing_field("rate_limited"))?,
-            authentication: authentication.ok_or_else(|| missing_field("authentication"))?,
+            rate_limited: if rate_limited.is_empty() {
+                Err(missing_field("rate_limited"))
+            } else {
+                Ok(rate_limited)
+            }?,
```

Comment:
I think this would be a bit more readable as

```suggestion
            rate_limited: if rate_limited.is_empty() {
                return Err(missing_field("rate_limited"));
            } else {
                rate_limited
            },
```